src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java

```diff
@@ -642,9 +642,23 @@ public ProjectBean getProject2(String projectIdentifier) {
       e.printStackTrace();
     }
     if (expDesign != null) {
-      // experimental design found and parsed. remove samples that have since been deleted:
+      // experimental design found and parsed. remove references to samples that have since been
+      // deleted; give openbis enough time to index samples
+      long creationTime = designExperiment.getRegistrationDetails().getModificationDate().getTime();
+      long currentTime = System.currentTimeMillis();
+      long elapsedTime = currentTime - creationTime;
+
+      long secondsInMilli = 1000;
+      long minutesInMilli = secondsInMilli * 60;
+      long hoursInMilli = minutesInMilli * 60;
+
+      long elapsedHours = elapsedTime / hoursInMilli;
```

Comment:
Increasing the readability by extracting this in an own small method, like `projectNeedsRefresh()` or sth